nntrainer/graph/network_graph.cpp

```diff
@@ -402,12 +415,12 @@ int NetworkGraph::realizeMultiOutputType(Layer &current) {
 }
 
 int NetworkGraph::isCompilable() {
-  if (layers.empty()) {
+  if (adj.empty()) {
     ml_loge("Layer is empty");
     return ML_ERROR_INVALID_PARAMETER;
   }
 
-  Layer &l = *layers[0];
+  Layer &l = *adj[0].front().layer;
```

Comment:
Is the first layer guaranteed to be the first layer?

Reply:
You are right. This is handled in a later commit of this PR. You can check the final version of this function as a check.